sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java

```diff
@@ -727,10 +730,35 @@ private boolean isExternalStorageMounted() {
     return os;
   }
 
+  @SuppressLint("NewApi") // we perform an if-check for that, but lint fails to recognize
   private void setAppPackageInfo(final @NotNull App app, final @NotNull PackageInfo packageInfo) {
     app.setAppIdentifier(packageInfo.packageName);
     app.setAppVersion(packageInfo.versionName);
     app.setAppBuild(ContextUtils.getVersionCode(packageInfo));
+
+    if (buildInfoProvider.getSdkInfoVersion() >= Build.VERSION_CODES.JELLY_BEAN) {
+      final Map<String, String> permissions = new HashMap<>();
+      final String[] requestedPermissions = packageInfo.requestedPermissions;
+      final int[] requestedPermissionsFlags = packageInfo.requestedPermissionsFlags;
+
+      if (requestedPermissions != null
+          && requestedPermissions.length > 0
+          && requestedPermissionsFlags != null
+          && requestedPermissionsFlags.length > 0) {
+        for (int i = 0; i < requestedPermissions.length; i++) {
+          String permission = requestedPermissions[i];
+          permission = permission.substring(permission.lastIndexOf('.') + 1);
+
+          final boolean granted =
+              (requestedPermissionsFlags[i] & REQUESTED_PERMISSION_GRANTED)
+                  == REQUESTED_PERMISSION_GRANTED;
+          permissions.put(permission, granted ? "granted" : "not_granted");
+        }
+      }
+      final Map<String, Object> unknown = new HashMap<>();
+      unknown.put("permissions", permissions);
```

Comment:
I'd rather add a new field to the App class, so it's easier for users to filter out or add data directly from the property instead of the `unknown` field, it's gonna render the same and ingest the same.
If you all have decided to go with the `unknown` field, also fine.

Reply:
so you mean we can add it only in the Android SDK, but ingest and fronted doesn't care right?

Reply:
Yes, if you add a new field to the data class, serialize/deserialize as `permissions`, everything should work normally.
`unknown` map is used only for not dropping unknown fields during serialization roundtrip, which is not the case right now.

Reply:
Changed to explicit field in https://github.com/getsentry/sentry-java/pull/2018/commits/7449b79f4e9239b3029ac500fcdf438212db0787